src/doc/rgw/multisite-reshard.md

```diff
@@ -71,6 +71,7 @@ The distinction between *index layout* and *log layout* is important, because in
 
 * Full sync uses a single bucket-wide listing to fetch all objects.
     - Use a cls_lock to prevent different shards from duplicating this work.
+        - Casey pointed out that this substantially reduces parallelism of full sync, and Yuval has recently noticed that it may be easy to induce a scenario where a large amount of ingest into a new bucket is being handled by full sync (with reduced perf);  could there be a way to prime new (and therefore empty) buckets for incremental sync on creation? 
```

Comment:
thanks for starting the discussion here @mattbenjamin 

also sharing a link to the email [rgw: draft of multisite resharding design for review](https://www.spinics.net/lists/dev-ceph/msg01310.html) for a little more context

two 'easy' suggestions i have here:
* increase concurrency of bucket full sync so it can sync ~100 objects at a time instead of 20, since we're not getting shard parallelism here
* when full sync is processing the list of remote objects, skip any whose mtime is higher than the time we started full sync - we can assume that we'll see any such entries in the bilog after switching to incremental sync, so there's no point in duplicating the sync here

longer-term, we might look for ways to run the bucket full sync in parallel with the per-shard bucket incremental syncs. i'm still not sure whether there are ordering requirements for sync of versioned objects, so that's one possible complication. the other complications are around how we track and report on the 'bucket sync status' when both of these things are happening at the same time